### PR TITLE
add extendable x-axis to hit efficiency plots (TRK DQM)

### PR DESCRIPTION
**Pull request by mtosi**
as discussed w/ @VinInn @makortel 
this PR is meant to extend to the hit efficiency plots the feature of having the x-axis extendable

### CODE REVIEW

**Comment by mtosi** [comment]
@cmsbuild, please test

**Comment by dmitrijus** [comment]
+1

**Comment by davidlange6** [comment]
hi @dmitrijus - you've previously claimed this feature (can extend) is a big problem for dqm. What has changed?

**Comment by dmitrijus** [comment]
@davidlange6 Because there was a bug in root, a strange interaction between kCanExtend and Alphanumeric histograms.
I believe it is fixed now. A quick check in root confirms, at least for non-alphanumeric histograms.

@mtosi However, why is this needed? Wouldn't it be wiser to set the limits to a constant?
TH1 code is horrible, no telling when it will break again... it better to limit the features DQM uses to a minimum.

**Comment by VinInn** [comment]
Because we need to support PU from 1 to  >200 and we cannot have histos with max 250 or even 300 just in case... (same for Lumi, LS etc)
TH1 is what CERN provides us since 20 (and more) years: unless a replacement (that will be called TH1) is provided we need to rely on that...

**Comment by dmitrijus** [comment]
>  we cannot have histos with max 250 or even 300 just in case...

Why not? I don't really understand... I think it is much much less confusion if all the histograms use the same binning.

I would understand if this was a number of bins / sparse histograms. But it isn't, kCanExtend does not change the number of bins (just extends/doubles the limits).

> TH1 is what CERN provides us since 20 (and more) years: unless a replacement (that will be anyhow called TH1) is provided we need to rely on that...

True. But we do care about performance and stability.
**kCanExtend** is bad for both performance (for fastest merging, binning has to be the same.... not entirely true, but this is what root itself tells us) and stability (track record :))

Anyway, I am not against it. If you or Mia tell me this is needed, I will give the +1.
It should not break anything, according to my best knowledge.

**Comment by mtosi** [comment]
I've just adjust the binning of the MEs as function of the # PVs
in addition, I've made the PV binning era dependent
phase1 :   40, -0.5,    79.5
phase2 : 125,  -0.5, 249.5

@VinInn @rovere @makortel  do you agree ?

**Comment by davidlange6** [comment]
please test

**Comment by dmitrijus** [comment]
+1

perfect!

File: DQM/TrackingMonitor/src/TrackingMonitor.cc [hunk 1]
```diff
@@ -500,17 +500,17 @@ void TrackingMonitor::bookHistograms(DQMStore::IBooker & ibooker,
...
 
-       double GoodPVtxMin   = conf->getParameter<double>("GoodPVtxMin");
-       double GoodPVtxMax   = conf->getParameter<double>("GoodPVtxMax");
+       double PVMin   = conf->getParameter<double>("PVMin");
+       double PVMax   = conf->getParameter<double>("PVMax");
        
        histname = "NumberOfGoodPVtxVsLUMI";
-       NumberOfGoodPVtxVsLUMI = ibooker.bookProfile(histname,histname,LUMIBin,LUMIMin,LUMIMax,GoodPVtxMin,3.*GoodPVtxMax,"");
+       NumberOfGoodPVtxVsLUMI = ibooker.bookProfile(histname,histname,LUMIBin,LUMIMin,LUMIMax,PVMin,3.*PVMax,"");
        NumberOfGoodPVtxVsLUMI->getTH1()->SetCanExtend(TH1::kAllAxes);
```
Comment (dmitrijus) [inline DQM/TrackingMonitor/src/TrackingMonitor.cc]: Hi, 
this is not touched by this PR, but can these be removed as well (in this or a separate pull request)?

File: DQM/TrackingMonitor/python/MonitorTrackInnerTrackMuons_cff.py [hunk 2]
```diff
@@ -45,9 +45,17 @@
...
-MonitorTrackMuonsInnerTrack.PVMin =   0.5
-MonitorTrackMuonsInnerTrack.PVMax = 120.5
+MonitorTrackMuonsInnerTrack.PVBin = 40
+MonitorTrackMuonsInnerTrack.PVMin = -0.5
+MonitorTrackMuonsInnerTrack.PVMax = 79.5 ## it might need to be adjust if CMS asks to have lumi levelling at lower values
+
+
+from Configuration.Eras.Modifier_phase1Pixel_cff import phase1Pixel
+from Configuration.Eras.Modifier_phase2_tracker_cff import phase2_tracker
+phase1Pixel.toModify(MonitorTrackMuonsInnerTrack, EtaBin=30, EtaMin=-3, EtaMax=3)
```
Comment (VinInn) [inline DQM/TrackingMonitor/python/MonitorTrackInnerTrackMuons_cff.py]: why 30 and not 31?

Comment (VinInn) [inline DQM/TrackingMonitor/python/MonitorTrackInnerTrackMuons_cff.py]: why 30 and not 31?

**Comment by mtosi** [comment]
@dmitrijus , do you like it, now ?
@VinInn, thanks ! I copied from what was used in other places ....

**Comment by mtosi** [comment]
@cmsbuild, please test

**Comment by jfernan2** [comment]
please test

**Comment by davidlange6** [comment]
+1

**Pull request merged**